### PR TITLE
Sdi 452 incentives api integration

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/SpringAPIServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Inherited
+@ExtendWith(
+  IncentivesApiExtension::class,
+  HmppsAuthApiExtension::class,
+)
+@ActiveProfiles("test")
+@EnableWebFlux
+@SpringBootTest(classes = [WebClientConfiguration::class, WebClientAutoConfiguration::class, OAuth2ClientAutoConfiguration::class, SecurityAutoConfiguration::class, ReactiveOAuth2ClientAutoConfiguration::class])
+@BootstrapWith(SpringBootTestContextBootstrapper::class)
+annotation class SpringAPIServiceTest
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/SpringAPIServiceTest.kt:39]: convenience annotation to bring minimal dependencies for a service test

File: README.md [hunk 2]
```diff
@@ -66,6 +66,18 @@ The CronJob calls the endpoint `/prisoner-index/queue-housekeeping` which is not
...
+
+*WE STRONGLY ADVISE, WHEN RUNNING LOCALLY, TO POINT AT SERVICES IN THE T3/DEV ENVIRONMENT*
+
+Since search has a number of dependencies that themselves have dependencies, it is recommended that when running locally you point mostly at services
+running in T3. The exception is pointing at localstack for AWS services, but HMPPS Auth, prison-api, incentives services are best used directly on T3 dev environment.
+The docker-compose.yml file is provided as reference but the dependencies of dependencies are not shown and are subject to change.
+To run against T3 dependencies the environment variables as shown in the helm_deploy/values-dev.yaml file for paths can be used.
+Personal client credentials for the T3 services can be obtained from the HMPPS Auth team with the appropriate roles.
+
+```shell
```
Comment (steverendell) [inline README.md:78]: not sure about these quotes?

**Review by steverendell — APPROVED** [review]
lgtm

File: README.md (hunk 2 above)
Comment (andymarke) [reply README.md:78]: Yes, cut pasts - I'll remove in next PR

**Pull request merged**